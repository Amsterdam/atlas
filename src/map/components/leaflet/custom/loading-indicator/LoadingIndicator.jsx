// import { Control, DomUtil } from 'leaflet';
import { MapControl } from 'react-leaflet';
import createLoadingIndicator from './LoadingInicatorHelper';
import './LoadingIndicator.scss';

export default class LoadingIndicator extends MapControl {

  constructor(props) {
    super(props);
    this.state = {
      map: this.props.map
    };
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.loading !== nextProps.loading && this.state.map) {
      if (nextProps.loading) {
        this.state.map.fire('dataloading');
      } else {
        this.state.map.fire('dataload');
      }
    }
  }

  shouldComponentUpdate() {
    return false;
  }

  componentDidMount() {
    super.componentDidMount();
  }

  createLeafletElement(props) { // eslint-disable-line class-methods-use-this
    const LoadingIndicatorControl = createLoadingIndicator();
    return new LoadingIndicatorControl(props);
  }
}
