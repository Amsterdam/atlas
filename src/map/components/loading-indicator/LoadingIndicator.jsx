import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const LoadingIndicator = ({ loading, showMapLink }) => {
  const loadingIndicatorClasses = classNames({
    'c-loading-indicator__container': true,
    'is-loading': loading,
    'c-loading-indicator__container--with-link': showMapLink,
  })
  return (
    <div className={`${loadingIndicatorClasses}`}>
      <div className="c-loading-indicator__box qa-loading-indicator">
        <img className="c-loading-indicator__icon" src="/assets/images/spinner.svg" alt="" />
        <span className="c-loading-indicator__text">Laden...</span>
      </div>
    </div>
  )
}

LoadingIndicator.defaultProps = {
  showMapLink: false,
}

LoadingIndicator.propTypes = {
  loading: PropTypes.bool.isRequired,
  showMapLink: PropTypes.bool,
}

export default LoadingIndicator
