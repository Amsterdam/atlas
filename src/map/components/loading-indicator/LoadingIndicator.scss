@import "../../../shared/styles/config";

.c-dashboard--page-type-data {
  .c-loading-indicator__container {
    right: $base-whitespace - 2;
  }
}

.c-loading-indicator__container {
  border: none;
  display: none;
  position: absolute;
  right: $base-whitespace * 5;
  top: $base-whitespace - 2;

  &.is-loading {
    display: block;
  }

  .c-loading-indicator__box {
    font-family: $base-font-family;
    font-size: $s-font-size;
    height: $base-whitespace * 3.5;
    left: 0;
    position: relative;
    top: 0;
    z-index: 999;
  }
}
