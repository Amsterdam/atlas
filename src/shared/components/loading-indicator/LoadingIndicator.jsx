import React from 'react'

import './_loading-indicator.scss'
import PropTypes from 'prop-types'
import { Spinner } from '@datapunt/asc-ui'

const LoadingIndicator = ({ IconComponent }) => (
  <div className="loading-indicator">{IconComponent}</div>
)

LoadingIndicator.defaultProps = {
  IconComponent: <Spinner size={36} color="secondary" />,
}

LoadingIndicator.propTypes = {
  IconComponent: PropTypes.node,
}

export default LoadingIndicator;
