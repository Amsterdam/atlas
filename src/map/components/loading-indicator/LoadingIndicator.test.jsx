import React from 'react'
import { shallow } from 'enzyme'
import LoadingIndicator from './LoadingIndicator'

describe('LoadingIndicator', () => {
  it('should render everything', () => {
    const wrapper = shallow(<LoadingIndicator loading={false} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with classname is-loadingf', () => {
    const wrapper = shallow(<LoadingIndicator loading />)
    expect(wrapper).toMatchSnapshot()
  })
})
