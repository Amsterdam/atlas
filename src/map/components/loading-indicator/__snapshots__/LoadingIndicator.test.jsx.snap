// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator should render everything 1`] = `
<div
  className="c-loading-indicator__container"
>
  <div
    className="c-loading-indicator__box qa-loading-indicator"
  >
    <img
      alt=""
      className="c-loading-indicator__icon"
      src="/assets/images/spinner.svg"
    />
    <span
      className="c-loading-indicator__text"
    >
      Laden...
    </span>
  </div>
</div>
`;

exports[`LoadingIndicator should render with classname is-loadingf 1`] = `
<div
  className="c-loading-indicator__container is-loading"
>
  <div
    className="c-loading-indicator__box qa-loading-indicator"
  >
    <img
      alt=""
      className="c-loading-indicator__icon"
      src="/assets/images/spinner.svg"
    />
    <span
      className="c-loading-indicator__text"
    >
      Laden...
    </span>
  </div>
</div>
`;
